apps/docs/docs/modules/high_level/chat_engine.md

```diff
@@ -11,7 +11,16 @@ const retriever = index.asRetriever();
 const chatEngine = new ContextChatEngine({ retriever });
 
 // start chatting
-const response = await chatEngine.chat(query);
+const response = await chatEngine.chat({ message: query });
+```
+
+The `chat` function also supports streaming, just add `stream: true` as an option:
+
+```typescript
+const stream = await chatEngine.chat({ message: query, stream: true });
+for await (const chunk of stream) {
+  process.stdout.write(chunk.response);
```

Comment:
@yisding @himself65 I am streaming here (and in `QueryEngine`) `Response` objects and store the delta in the `response` string.

We might alternatively:
1. Use a dedicated interface (e.g. `ResponseChunk`) containing the deltas (like we're doing it for the LLM interface)
2. Extend the `Response` interface with a `delta` 

What are your thoughts?

Reply:
Also, we might want to rename the `message` parameter. This was the original name, but now that it's a named parameter, it feels kind of weird, as it's just a `string` and not a message object.

Reply:
> we might want to rename the `message` parameter

use `text`?

Reply:
Actually, it's not just `text` - we also have multi-modal messages. 

see a multi-modal usage:
https://github.com/run-llama/LlamaIndexTS/blob/ab9beefa0d18ef8dc775a5d1c5d253828d16df05/packages/create-llama/templates/types/streaming/nextjs/app/api/chat/route.ts#L51-L62

Reply:
The precise naming would be `userMessageContent`, which is too wordy, and I think just `content` is a bit confusing. Let's stick with `message` for now.